.github/workflows/nightly.yml

```diff
@@ -0,0 +1,67 @@
+name: Publish Nightly Builds
+
+on:
+  push:
+    branches:
+      - nightly_build
+
+jobs:
+  build-test-publish:
+    strategy:
+      matrix:
+        include:
+          - runson: macos-latest # mac
+
+          - runson: ubuntu-latest # linux x64
+            image: docker://public.ecr.aws/async-profiler/asprof-builder-x86:latest
```

Comment:
Can we handle both x64 and arm64 builds uniformly? I mean, this can also be moved to CodeBuild.

Reply:
Possible, but codebuild does take a while longer to execute

Reply:
https://github.com/openorclose/async-profiler/actions/runs/11016650956 suggests the opposite, but maybe it's because of a cached image.

It's worth noting that the original asbrof-builder image based on ubuntu:18 is only 292 MB, while your asprof-builder-x86 is 4.25 GB. Why is it so large and is it possible to shrink it?

Reply:
have reduced the docker image size to <400MB, and made it as similar as possible to the original one